task/backend/executor/task_executor_test.go

```diff
@@ -19,31 +20,56 @@ import (
 	"github.com/influxdata/influxdb/query"
 	"github.com/influxdata/influxdb/task/backend"
 	"github.com/influxdata/influxdb/task/backend/scheduler"
+	"github.com/opentracing/opentracing-go"
+	"github.com/uber/jaeger-client-go"
 	"go.uber.org/zap/zaptest"
 )
 
+func TestMain(m *testing.M) {
+	var code int
+	func() {
+		// backup old tracer
+		oldTracer := opentracing.GlobalTracer()
+		defer opentracing.SetGlobalTracer(oldTracer)
+
+		// setup in-memory tracer for task tests
+		sampler := jaeger.NewConstSampler(true)
+		tracer, closer := jaeger.NewTracer("task_backend_tests", sampler, jaeger.NewInMemoryReporter())
+		defer closer.Close()
+
+		opentracing.SetGlobalTracer(tracer)
+
+		code = m.Run()
+	}()
+
+	os.Exit(code)
```

Comment:
note for reviewer: because of the nature of opentracing using a global variable for the tracer + the fact we use `t.Parallel()` everywhere I was triggering race conditions when setting this tracer in the specific tests I needed it.
I moved it up here to ensure this happens once at the start and then no tracer setting shenanigans could trigger a race later on.